core/src/main/java/io/cloudevents/core/impl/BaseCloudEventBuilder.java

```diff
@@ -43,6 +46,32 @@ public BaseCloudEventBuilder() {
         this.extensions = new HashMap<>();
     }
 
+    public BaseCloudEventBuilder(CloudEventContext context) {
+        this();
+        for (String name: context.getAttributeNames()) {
+            if (!name.equals("specversion")) {
+                Object value = context.getAttribute(name);
+                if (value instanceof String) {
+                    withAttribute(name, (String) value);
+                } else if (value instanceof URI) {
+                    withAttribute(name, (URI) value);
+                } else if (value instanceof OffsetDateTime) {
+                    withAttribute(name, (OffsetDateTime) value);
+                }
+            }
+        }
+        for (String name: context.getExtensionNames()) {
+            Object value = context.getExtension(name);
+            if (value instanceof String) {
+                withExtension(name, (String) value);
+            } else if (value instanceof Number) {
+                withExtension(name, (Number) value);
+            } else if (value instanceof Boolean) {
+                withExtension(name, (Boolean) value);
+            }
+        }
+    }
```

Comment:
Can't you tweak a bit `setAttributes` for that? You don't really need new code here i guess

Reply:
`setAttributes` is taking the `CloudEvent` just because we didn't had a unique interface for attributes and extensions, but it's doing pretty much what you're doing here

Reply:
`setAttributes()` takes a `CloudEvent` and delegates to some other code. I don't see an obvious way to adapt that to a `CloudEventContext`. If you can suggest something, go ahead.

Reply:
I.e. `setAttributes()` delegates to a `CloudEventReaderAdapter` and that can only be created with a `CloudEvent`.

Reply:
I can add another adapter I guess. Trying that now...

Reply:
See ecfab75